In [1]:
from scipy import stats
import pandas as pd

In [2]:
import numpy as np

### A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29


## n=15, sample_mean(x)=1.23

In [9]:
data=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

In [11]:
data_mean=np.mean(data)          # to find mean value of given dataset
print(data_mean)

1.2386666666666666


In [13]:
std_dev=np.std(data)              # to find sample standard deviation of given dataset
print(std_dev)

0.18661427836285438


In [18]:
inter=stats.t.interval(0.99,loc=1.2386,scale=0.1866/np.sqrt(15), df=14)          
inter

(np.float64(1.0951759765588598), np.float64(1.38202402344114))

### Here i found t-distribution because total number sample is less than 30 and population standard deviation is unknown,
### 99% confidence interval is between 1.0951 to 1.3820

In [19]:
inter1=stats.norm.interval(0.99,loc=1.2389,scale=0.2/np.sqrt(15))
inter1

(np.float64(1.105884746729094), np.float64(1.3719152532709058))

### Here i took normal z-distribution because population standard deviation is known,
### Here i found 99% of confidence interval between 1.0951 million characters to 1.3820 characters.